doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -571,25 +562,31 @@ Chapter :ref:`thecoqcommands` for documentation on how to compile a file). A com
 file is a particular case of module called *library file*.
 
 
-.. cmd:: Require @qualid
+.. cmd:: Require {? {| Import | Export } } {+ @qualid }
 
-   This command looks in the loadpath for a file containing module :n:`@qualid`
-   and adds the corresponding module to the environment of |Coq|. As
-   library files have dependencies in other library files, the command
-   :cmd:`Require` :n:`@qualid` recursively requires all library files the module
-   qualid depends on and adds the corresponding modules to the
-   environment of |Coq| too. |Coq| assumes that the compiled files have been
-   produced by a valid |Coq| compiler and their contents are then not
-   replayed nor rechecked.
+   Loads compiled modules into the |Coq| environment.  The command
+   searches the loadpath for the modules named by the :n:`@qualid`\s.  Each :n:`@qualid`
+   has the form :n:`@dirpath.@ident`.  The commmand maps :n:`@dirpath` to a physical
+   directory (see Section :ref:`libraries-and-filesystem`) and loads the compiled
+   file :n:`@ident.vo` from that directory.  (Compiling :n:`@ident.v` generates :n:`@ident.vo`.)
 
-   To locate the file in the file system, :n:`@qualid` is decomposed under the
-   form :n:`dirpath.@ident` and the file :n:`@ident.vo` is searched in the physical
-   directory of the file system that is mapped in |Coq| loadpath to the
-   logical path dirpath (see Section :ref:`libraries-and-filesystem`). The mapping between
+   The process is applied recursively to all the loaded modules.
+   The compiled files must have been compiled with the same version of |Coq|.
+   The compiled files are neither replayed nor rechecked.
+
+   * :n:`Import` - TBA
+   * :n:`Export` - TBA
+
+   .. todo: I can't make sense of the old explanations for Import and Export
```

Comment:
It's rather simple but the explanation is convoluted. Library files are modules. Requiring a library file makes the module available. Using `Require Import` is pretty much the same as a `Require` followed by an `Import` on the module name: it makes the objects contained in the module available by their short name. `Export` does the same thing as `Import` but *in addition* it makes the objects that were loaded available to any module that `Import` the current module. Cf. https://coq.github.io/doc/master/refman/language/gallina-extensions.html#coq:cmd.import and https://coq.github.io/doc/master/refman/language/gallina-extensions.html#coq:cmd.export.

Reply:
And I gather that Import and Export are not recursive the was Require is?  (and if not, maybe clarify that in Import/Export)

Reply:
I think I got it.

Reply:
Indeed, `Require` is recursive. As for `Export`, we can say it is "transitive" i.e. if you have a chain of `Require Export`, then you get all the objects at the end, because each time, they have been imported and re-exported.

Reply:
Got it, without mentioning "transitive".